File: src/Dota2Analyzer.Core/Models/OpenDota/ItemConstants.cs
using System.Text.Json.Serialization;

namespace Dota2Analyzer.Core.Models.OpenDota;

public sealed record ItemConstants
{
    [JsonPropertyName("id")] public int? Id { get; init; }
    [JsonPropertyName("img")] public string Img { get; init; } = string.Empty;
    [JsonPropertyName("dname")] public string DisplayName { get; init; } = string.Empty;
    [JsonPropertyName("cost")] public int? Cost { get; init; }
    [JsonPropertyName("qual")] public string Quality { get; init; } = string.Empty;
    [JsonPropertyName("components")] public List<string>? Components { get; init; }
}

File: dota2-analyzer-common/src/main/java/com/dota2analyzer/core/model/opendota/ItemConstants.java
package com.dota2analyzer.core.model.opendota;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class ItemConstants {

    @JsonProperty("id")
    private Integer id;

    @JsonProperty("img")
    private String img = "";

    @JsonProperty("dname")
    private String displayName = "";

    @JsonProperty("cost")
    private Integer cost;

    @JsonProperty("qual")
    private String quality = "";

    @JsonProperty("components")
    private List<String> components;

    public ItemConstants() {}

    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }

    public String getImg() { return img; }
    public void setImg(String img) { this.img = img; }

    public String getDisplayName() { return displayName; }
    public void setDisplayName(String displayName) { this.displayName = displayName; }

    public Integer getCost() { return cost; }
    public void setCost(Integer cost) { this.cost = cost; }

    public String getQuality() { return quality; }
    public void setQuality(String quality) { this.quality = quality; }

    public List<String> getComponents() { return components; }
    public void setComponents(List<String> components) { this.components = components; }
}
